.github/workflows/build_uboot.yml

```diff
@@ -0,0 +1,191 @@
+name: Build U-Boot for Xilinx Platforms
+# This workflow builds U-Boot for various Xilinx platforms.
+
+on:
+  workflow_dispatch:
+    inputs:
+      vivado_version:
+        description: 'Vivado version that will be used for download xilinx versal'
+        required: true
+        default: '2023.2'
+      x32_defconfigs:
+        description: '32-bit configurations to build'
+        required: true
+        default: 'zynq_zc702_defconfig zynq_zc706_defconfig zynq_zed_defconfig zynq_picozed_defconfig zynq_coraz7_defconfig zynq_adrv9361_defconfig'
+      export_on_server:
+        description: 'Export on server'
+        type: boolean
+        default: false
+
+permissions:
+  id-token: write
+  contents: write
+  actions: read
+
+env: 
+  GCC_BASE_FOLDER: /usr/bin
+  LINARO_FOLDER: /opt/linaro
+  uboot_checkout: u-boot-xlnx
+  ARTIFACTS_NAME: u-boot-artifacts
+  CLOUDSMITH_NAMESPACE: adi
+
+jobs:
+  build-uboot:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install GCC 
+        run: |
+          wget https://releases.linaro.org/components/toolchain/binaries/7.5-2019.12/arm-linux-gnueabihf/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz
+          sudo mkdir -p ${{env.LINARO_FOLDER}}
+          sudo tar -xf gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz -C /opt/linaro
+          
+          sudo apt update
+          sudo apt install -y gcc-aarch64-linux-gnu
+          
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          repository: analogdevicesinc/u-boot-xlnx
+          ref: master
+          path: ${{env.uboot_checkout}}
+      - name: Create artifacts folder
+        run: |
+          rm -rf ${{env.ARTIFACTS_NAME}}
+          mkdir -p ${{env.ARTIFACTS_NAME}}
+      
+      - name: Run 32 bit configurations
+        working-directory: ${{env.uboot_checkout}}
+        env:
+          GCC_LINARO_FOLDER: ${{env.LINARO_FOLDER}}//gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf/bin
+          ARCH: arm
+          X32_DEFCONFIGS: "${{ inputs.x32_defconfigs }}"
+        run: |
+          export CROSS_COMPILE=${{env.GCC_LINARO_FOLDER}}/arm-linux-gnueabihf-
+          export ARCH=${{env.ARCH}}
+
+          for defcfg in ${{env.X32_DEFCONFIGS}}; do
+             make distclean
+             make $defcfg
+             make -j2
```

Comment:
You can build with maximum number of threads (we don't need to limit anymore)...